### ISSUE TITLE
Refactor: Unique tokens

### ISSUE THREAD

**Issue by peterzeller** [SyncFree/antidote_crdt#11]
Many modules include the following function:

```
-spec unique() -> token().
unique() ->
    crypto:strong_rand_bytes(20).
```

I would extract this into a separate module and search for a replacement for `crypto:strong_rand_bytes`, because the documentation for the function says:

> May throw exception low_entropy in case the random generator failed due to lack of secure "randomness".

I don't think we need secure random numbers in our case.

@gyounes @vitorenesduarte do you want to work on this?

**Comment by PascalDierich** [comment]
All tokens are created in the `downstream` function which takes the `update`-op and the CRDT itself as arguments.

Would it maybe be sufficient to concat this two plus the current `timestamp` and hash it?
The `update`-ops are of course not random at all, the CRDT-state definitely not enough.

The question would be: Can there be two concurrent same update operations with the same CRDT-state at the exact same time (or at least same timestamp)?

Otherwise I guess we dont have enough randomness in the `downstream` function to not rely on the OS entropy pool.

**Comment by peterzeller** [comment]
I was rather thinking of something simple like `rand:uniform`.

I would not use hashes of the updates or the state as that depends on the size and could be slower than `strong_rand_bytes`. 

So far we did not have any problems with `strong_rand_bytes`. I guess problems would only occur on embedded devices without a good entropy source. It might be that `rand:uniform` is a bit faster on normal servers, but I don't know if that would make a big difference. I compared the runtime of the two functions in the Erlang shell and it was only 20% faster there, so probably not worth the change. 

The best solution would probably be to generate some form of identifiers on the Antidote side, but that would be a bigger change.
